### PR TITLE
:book: update supported Kubernetes versions

### LINKED ISSUE #6155: Use socket with unix:/// in CAPD.
We are currently using `/var/run/containerd/containerd.sock` in our CAPD configurations.

We should actually be using `unix:///var/run/containerd/containerd.sock`. Goal of this issue is to go over test/ and replace all occurrences.

P.S. there will be a warning with kubeadm v1.24 for paths which are not using a URL scheme. (xref: https://github.com/kubernetes/kubeadm/issues/2426#issue-846676028)

/kind cleanup
/area provider-docker
/good-first-issue

[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Update supported versions in the book to include 1.24.
I refactored the table a bit to improve readability + to be able to include v1.24 without adding another two columns.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #5968
Fixes #6155

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/retitle :book: update supported Kubernetes versions
/lgtm

Aside - not a  blocker - are we planning to drop 0.3/0.4/1.0 from the main version of the book ahead of the next release?

**Comment by sbueringer** [comment]
I'll use this PR to test some jobs (in addition to testgrid as the update test there run only once per day):

/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

**Comment by sbueringer** [comment]
> Aside - not a blocker - are we planning to drop 0.3/0.4/1.0 from the main version of the book ahead of the next release?

I don't know. On one side those versions are all out of support. On the other side we only kept the version page of the current book up-to-date. This means that we might lose the information which Kubernetes version we support in which CAPI version if we just drop it here (not sure, we can verify the old books).

**Comment by sbueringer** [comment]
/hold cancel
/assign @fabriziopandini 
Test failure is obviously unrelated and will be fixed elsewhere.

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -121,6 +124,16 @@ The Kubeadm Control Plane requires the Kubeadm Bootstrap Provider.
 | v1.1 (v1beta1)  | v1.8.6                          |
 | v1.2 (v1beta1)  | v1.9.2                          |
 
+#### Kubernetes version specific notes
+
+**1.24**:
+
+* Kubeadm Bootstrap Provider:
+    * `kubeadm` now sets both the `node-role.kubernetes.io/control-plane` and `node-role.kubernetes.io/master` taints on control plane nodes.
+    * `kubeadm` now only sets the `node-role.kubernetes.io/control-plane` label on control plane nodes (`node-role.kubernetes.io/master` is not set anymore).
```
Comment (chrischdi) [inline docs/book/src/reference/versions.md:133]: Does it now set both or only :-)

Comment (sbueringer) [reply docs/book/src/reference/versions.md:133]: Both taints only the control plane label

Comment (sbueringer) [reply docs/book/src/reference/versions.md]: Added another "label" in the second point to avoid confusion

Comment (chrischdi) [reply docs/book/src/reference/versions.md]: ah, was a bit confusing to find the distinction 👍 

Maybe rephrasing it to have `taints` and `label` not at the end would make it more readable.

But also ok for me as is.

Comment (chrischdi) [reply docs/book/src/reference/versions.md]: Or highlighting tains/label in some way.

Comment (sbueringer) [reply docs/book/src/reference/versions.md]: I would rely now on folks reading it twice if they are confused :)

**Comment by enxebre** [comment]
@sbueringer are we currently running tests for kube >= 1.18?

**Comment by sbueringer** [comment]
> @sbueringer are we currently running tests for kube >= 1.18?

We have one test where we install v1.18, then upgrade to v1.19 and then run conformance tests.

Most versions that we claim we have support for are only tested by upgrade tests. (https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api).

We only run more tests for v1.19 (min) and v1.24 (max) management clusters (e.g. capi-e2e-main uses v1.24, capi-e2e-mink8s-main uess v1.19)

**Pull request merged**